oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/authorization/OAuth2AuthorizationServerConfigurer.java

```diff
@@ -217,6 +230,16 @@ public void configure(B builder) {
 			builder.addFilterBefore(postProcess(oidcProviderConfigurationEndpointFilter), AbstractPreAuthenticatedProcessingFilter.class);
 		}
 
+		RegisteredClientRepository registeredClientRepository = getRegisteredClientRepository(builder);
+		OAuth2AuthorizationService authorizationService = getAuthorizationService(builder);
+
+		OidcClientRegistrationEndpointFilter oidcClientRegistrationEndpointFilter =
+				new OidcClientRegistrationEndpointFilter(
+						registeredClientRepository,
+						authorizationService,
+						providerSettings.oidcClientRegistrationEndpoint());
+		builder.addFilterBefore(postProcess(oidcClientRegistrationEndpointFilter), FilterSecurityInterceptor.class);
```

Comment:
`OidcClientRegistrationEndpointFilter` should be after `FilterSecurityInterceptor` - ensure there is a test for this. `OAuth2TokenEndpointFilter` and `OAuth2TokenRevocationEndpointFilter` are also after `FilterSecurityInterceptor` since these 3 `Filter` need to be protected

Reply:
Fixed (it was a mistake from my side, the intention was to add it after the `FilterSecurityInterceptor` )